extensions/azurePublish/src/components/azureProvisionDialog.tsx

```diff
@@ -50,6 +48,18 @@ import { ChooseResourcesList } from './ChooseResourcesList';
 import { getExistResources, removePlaceholder, decodeToken, defaultExtensionState } from './util';
 import { ResourceGroupPicker } from './ResourceGroupPicker';
 
+type ProvisionFormData = {
+  choice: string;
+  tenantId: string;
+  subscriptionId: string;
+  resourceGroup: string;
+  hostname: string;
+  region: string;
+  luisLocation: string;
+  enabledResources: ResourcesItem[];
```

Comment:
This looks a lot cleaner.  Nit: If we are going to clean this up, then I think we eventually need to have the list of possible resources (required and optional) outside this type and only have the list of selected resource IDs as form data.